source/retryable-writes/tests/insertMany.yml

```diff
@@ -45,12 +45,12 @@ tests:
     -
         description: "InsertMany fails after multiple network errors"
         failPoint:
-            mode: { times: 2 }
+            mode: "alwaysOn"
```

Comment:
I won't argue that this fixes the test, but do we know why? If `failBeforeCommitExceptionCode` is set, the write should never be committed, so I would expect 2 fail point triggers would be enough to prevent the retryable write from succeeding.

Reply:
Sorry, I do not know why. I'll ask Kal.

Reply:
Quoting @jsmulrow from Slack:

> So it looks like if the server isn’t able to insert the documents as a batch, it will try to insert each document one at a time: https://github.com/mongodb/mongo/blob/f96e0ef10baac1d2ca254c2c4ef78ef0ff8a0542/src/mongo/db/ops/write_ops_exec.cpp#L387-L388
> So my guess is that the batched attempt fails, then the server tries each document individually, which exhausts the failpoint. So then when the command is retried, the failpoint is inactive.

In this case, we may be blowing through `times=2` in the first write command. `times=3` may be the conservative fix. That said, while I see value in crafting the test to exercise the boundary condition, one could argue that is more appropriate for the server's test suite and we should stick with the most resilient option of "alwaysOn".

Reply:
I stuck with "alwaysOn" and added a comment.